### ISSUE TITLE
BUG tostring / coalesce and coalesce / tostring

### ISSUE THREAD

**Issue by davidhbigelow** [neo4j/neo4j#7718]
- Neo4j version: 3.0.4
- Operating system: ALL
- API/Driver: Cypher
- **Steps to reproduce**
- Expected behavior -- Same as 2.X
- Actual  - errors are thrown - fix requires a retooling of old cypher code generation! :(

**Neo4j Version:**  3.0.4
**Operating System:** ALL
**API:** Cypher
### Steps to reproduce

```
// 2.3.5 syntax - WORKS!
with ["male","female",null] as gender
unwind gender as gen
return coalesce(str(gen),"x") as result
```

```
// 2.3.5 syntax - WORKS!
with ["male","female",null] as gender
unwind gender as gen
return str(coalesce(gen,"x")) as result
```

```
// 3.0.4 syntax - FAILS!
with ["male","female",null] as gender
unwind gender as gen
return coalesce(tostring(gen),"x") as result
```

```
// 3.0.4 syntax - FAILS!
with ["male","female",null] as gender
unwind gender as gen
return tostring(coalesce(gen,"x")) as result
```
### Expected behavior

Allows a string-coalesce or a coalesce-string nesting of the function calls.
### Actual behavior

```
Type mismatch: expected Boolean, Float, Integer or String but was Any, String (line 3, column 26 (offset: 84))
"return coalesce(tostring(gen),"x") as result"
```

**Comment by spacecowboy** [comment]
Thanks for the report

**Comment by bhativ1303** [comment]
Its happening in 3.1 as well.

**Comment by bhativ1303** [comment]
I will pick it up.

**Comment by Mats-SX** [comment]
Fixed by #7780.
